_announcements/week-04.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Week 4 Review 
+week: 4
+date: 2023-02-3
+---
+
+# 🍀  Week 4: Making decisions in programs (Branching)
+
+Dear all,
+
+We’re almost halfway through the quarter!
+
+- During class and lab, we discussed different data structures such as:
+    - Conditionals
+    - Booleans and Boolean Operators: Comparison Operators and Logical Operators
+    - Branching using the `if`-`elif`-`else` statements
+    - `assert` statements for writing tests of return values
+        
+* The slides and class recordings as well as the videos from the class and lab sections are posted on Gauchospace (look for the ⏩ icon; use your <span style="color:red">**UCSB address**</span> - we won't be granting access to personal gmail accounts.)
+
+* The last section of every chapter in zyBook contains our class notes, which we update afterward with whatever was discussed in class. Sometimes, like this week, there is additional code for you to practice with (the number-guessing game using the `random` module).
+
+## ⏰ Week 5 - Due dates and Deadlines ⏰
+
+Check the [Calendar]({{site.url}}/{{site.baseurl}}/calendar#week-4), for the due dates. They are the same as they've been before.
+
+Upcoming deadlines are:
+* ⏰  **Tuesday, 2/07 11:59PM**: **LA04**
+* ⏰  **Wednesday, 2/08 11:59PM**: **LA05 CHECKPOINT** Week 5 zyLabs with the word “Checkpoint” in their title
+* ⏰ **Thursday, 2/09**: in-class Quiz 4 on Gauchospace, covering Week 1-4 content.
+* ⏰  **Saturday, 2/11 11:59PM**: Weekly reflection - the link to it is available under the **Week 4 tab** on Gauchospace + **PA05** + **CA05**
+
+Important: **Next Monday, 2/06** is the deadline to drop this course without a W. 
+> <span style="color:orange"> Keep in mind that the College of Engineering [does not allow this course to be switched to P/NP later]({{site.url}}/{{site.baseurl}}/about#note-about-pnp).</span>
+
+
+## ➡️    Staying on Track
+
+* Ask for clarifications during the [office hours]({{site.url}}/{{site.baseurl}}/schedule/), lab sessions, or post your questions on [Piazza]({{site.aux_links.Piazza}})
+
+🔮 [Looking ahead to Week 5]({{site.url}}/{{site.baseurl}}/calendar#week-5), your tasks for the upcoming days are:
+* Carefully read the instructions! Do NOT rush through them. 💎
+    - Practice writing reusable and generalizable code using functions.
```

Comment:
Take non-function code from zyBooks and see if you can turn it into a function. 
Write tests for it.
Feel free to ask us during the office hours if you are on the right track.